udata/search/fields.py

```diff
@@ -145,23 +148,30 @@ def __init__(self, field, model, labelizer=None, field_name='id'):
         self.model = model
         self.field_name = field_name
 
-    def from_response(self, name, response):
-        is_objectid = isinstance(getattr(self.model, self.field_name),
-                                 db.ObjectIdField)
-        cast = lambda o: ObjectId(o) if is_objectid else o
-
+    def from_response(self, name, response, fetch=True):
         facet = response.get('facets', {}).get(name)
         if not facet:
             return
         ids = [term['term'] for term in facet['terms']]
-        if is_objectid:
-            ids = map(ObjectId, ids)
-        objects = self.model.objects.in_bulk(ids)
+
+        if fetch:
```

Comment:
The fetch part is the existing code (model are fetched to render the facets in front).
The only ne part is the `else` part which only returns the model class and its identifier.